dev/com.ibm.ws.security.mp.jwt_fat/fat/src/com/ibm/ws/security/mp/jwt/fat/MPJwtConfigUsingBuilderTests.java

```diff
@@ -0,0 +1,1087 @@
+        serverTracker.addServer(server);
+        server.startServerUsingExpandedConfiguration(configFile);
+        saveServerPorts(server, MpJwtFatConstants.BVT_SERVER_1_PORT_NAME_ROOT);
+        server.addIgnoredErrors(Arrays.asList(MpJwtMessageConstants.CWWKW1001W_CDI_RESOURCE_SCOPE_MISMATCH));
+    }
+
+    //    protected static void setUpAndStartBuilderServer(LibertyServer server, String configFile, boolean jwtEnabled, String builderApp) throws Exception {
+    //        deployBuilderServerComplexBuilderApp(server);
+    //        setUpAndStartBuilderServer(server, configFile, jwtEnabled);
+    //    }
+
+    /**
+     * Initialize the list of test application urls and their associated classNames
+     * 
+     * @throws Exception
+     */
+    protected static void initTestAppArray() throws Exception {
+
+        //List<List<String>>
+        testApps = new ArrayList<List<String>>();
+        testApps.add(Arrays.asList(buildAppUrl(resourceServer, MpJwtFatConstants.MICROPROFILE_SERVLET, MpJwtFatConstants.MPJWT_APP_SEC_CONTEXT_REQUEST_SCOPE),
+                                   MpJwtFatConstants.MPJWT_APP_CLASS_SEC_CONTEXT_REQUEST_SCOPE));
+        testApps.add(Arrays.asList(buildAppUrl(resourceServer, MpJwtFatConstants.MICROPROFILE_SERVLET, MpJwtFatConstants.MPJWT_APP_TOKEN_INJECT_REQUEST_SCOPE),
+                                   MpJwtFatConstants.MPJWT_APP_CLASS_TOKEN_INJECT_REQUEST_SCOPE));
+        testApps.add(Arrays.asList(buildAppUrl(resourceServer, MpJwtFatConstants.MICROPROFILE_SERVLET, MpJwtFatConstants.MPJWT_APP_CLAIM_INJECT_REQUEST_SCOPE),
+                                   MpJwtFatConstants.MPJWT_APP_CLASS_CLAIM_INJECT_REQUEST_SCOPE));
+
+    }
+
+    /**
+     * All of the tests in this class follow the same flow. The differences between them are which builder they use to create a
+     * token, the config they use in the resource server
+     * and then whether they expect a failure (mainly due to a mis-match between the token and the servers config).
+     * We'll put the common steps in this method so we're not duplicating steps/code over and over.
+     * 
+     * @param builtToken
+     *            - the token built to reflect the goal of the calling test
+     * @param expectations
+     *            - the expected behavior that we need to validate
+     * @throws Exception
+     */
+    public void genericConfigTest(String builtToken) throws Exception {
+        genericConfigTest(builtToken, null);
+    }
+
+    public void genericConfigTest(String builtToken, Expectations expectations) throws Exception {
+
+        JwtTokenTools jwtTokenTools = new JwtTokenTools(builtToken);
+
+        jwtTokenTools.printJwtContent();
+
+        WebClient webClient = actions.createWebClient();
+        boolean setGoodExpectations = false;
+        if (expectations == null) {
+            setGoodExpectations = true;
+        }
+        for (List<String> app : testApps) {
+            if (setGoodExpectations) {
```

Comment:
The `setGoodExpectations` variable is a bit superfluous - this `if` statement could just be `if (expectations == null)` without sacrificing clarity.